services/121-service/src/registration/dto/registration-status-patch.dto.ts

```diff
@@ -1,6 +1,7 @@
 import { ApiProperty } from '@nestjs/swagger';
-import { IsEnum, IsOptional, IsString, MinLength } from 'class-validator';
+import { IsEnum, IsOptional } from 'class-validator';
 import { RegistrationStatusEnum } from '../enum/registration-status.enum';
+import { IsNotBothPresent } from '../validators/is-not-both-present.class.validator';
 
 export class RegistrationStatusPatchDto {
```

Comment:
I think the addition of these validators to the DTOs should be covered by unit tests to make sure we don't regress.

Something like
- it should throw error if both are defined
- it should throw error if neither are defined
- it should work fine if only message is defined
- it should work fine if only messageTemplateKey is defined

Reply:
Good point. I'll make it depending on this thread: https://github.com/global-121/121-platform/pull/5264#discussion_r1593784931